File: Kattis/wertyu.hs
-- https://open.kattis.com/problems/wertyu
--
{-# LANGUAGE FlexibleInstances #-}
{-# LANGUAGE FunctionalDependencies #-}
{-# LANGUAGE MultiParamTypeClasses #-}
{-# OPTIONS_GHC -O2 #-}

import System.IO (isEOF)

a :: String
a = "1234567890-=WERTYUIOP[]\\SDFGHJKL;'XCVBNM,./ "

b :: String
b = "‘1234567890-QWERTYUIOP[]ASDFGHJKL;ZXCVBNM,. "

elemIndex :: Char -> String -> Int
elemIndex ch s
  | notElem ch s = -1
  | otherwise = minimum [i | (i, j) <- zip [0 ..] s, j == ch]

solve :: String -> String
solve s = [f ch | ch <- s]
  where
    f :: Char -> Char
    f c = if elem c a then b !! (elemIndex c a) else c

main :: IO ()
main = do
  read_eof <- isEOF
  if read_eof
    then return ()
    else do
      s <- getLine
      putStrLn (solve s)
      main

File: 校招Collection/Others/WERTYU.py
# https://www.nowcoder.com/practice/0f318e6fffe3490cb700e16b65a93b1b

a = "1234567890-=WERTYUIOP[]\SDFGHJKL;'XCVBNM,./"
b = "`1234567890-QWERTYUIOP[]ASDFGHJKL;ZXCVBNM,."

table = dict(zip(a, b))
try:
    while 1:
        s = raw_input()
        output = ''
        for i in s:
            if i in table:
                output += table[i]
            else:
                output += i
        print output
except:
    pass